core/zoom_controls.js

```diff
@@ -302,7 +302,11 @@ Blockly.ZoomControls.prototype.createZoomResetSvg_ = function(rnd) {
   Blockly.bindEventWithChecks_(zoomresetSvg, 'mousedown', null, function(e) {
     ws.markFocused();
     ws.setScale(ws.options.zoomOptions.startScale);
+    ws.beginCanvasTransition();
     ws.scrollCenter();
+    setTimeout(function() {
+      ws.endCanvasTransition();
+    }, 500);
```

Comment:
Add transition to zoom-in and zoom-out buttons?

Reply:
Filed as #2023 after discussion--using the same code doesn't make it look smooth with multiple zooms.